common/persistence/persistence_rate_limited_clients.go

```diff
@@ -1115,8 +1180,11 @@ func allow(
 	if ok := namespaceRateLimiter.Allow(now, quotaRequest); !ok {
 		return ErrPersistenceNamespaceLimitExceeded
 	}
+	if ok := shardRateLimiter.Allow(now, quotaRequest); !ok {
```

Comment:
I think to match previous behavior, this limiter should go before `namespaceRateLimiter`

It is technically a per-ns+per-shard limit and it is the most restrictive and should be checked first to avoid issues in the golang rate limiter with requesting and canceling tokens very quickly.

Reply:
Makes sense. I have updated the order.